### PR TITLE
#65 Skelton CLI for DwcaValidationFinishedMessage Handling

### PR DESCRIPTION
**Pull request by aalbatross**
Skelton CLI code listening to DwCADownloadFinishedMessage.
Issue:
https://github.com/gbif/pipelines/issues/65

### CODE REVIEW

File: crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroConverterCallBack.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Call back which is called when the {@link org.gbif.common.messaging.api.messages.DwcaDownloadFinishedMessage } is received.
+ * ToDo Implement DwCAToAvro Conversion
+ */
+public class DwCAToAvroConverterCallBack extends AbstractMessageCallback<DwcaDownloadFinishedMessage> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DwCAToAvroConverterCallBack.class);
+
+  @Override
+  public void handleMessage(DwcaDownloadFinishedMessage dwcaDownloadFinishedMessage) {
+    LOG.info("Received Download finished message, " + dwcaDownloadFinishedMessage.toString());
```
Comment (timrobertson100) [inline crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroConverterCallBack.java]: This should listen for the validation report, not the download finished.  This is documented in the spec on https://github.com/gbif/pipelines/issues/59

File: crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroConverterCallBack.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package org.gbif.crawler.dwca.avroconverter;
+
+import org.gbif.common.messaging.AbstractMessageCallback;
+import org.gbif.common.messaging.api.messages.DwcaDownloadFinishedMessage;
```
Comment (fmendezh) [inline crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroConverterCallBack.java]: unused import

File: crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroConverterCallBack.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * Call back which is called when the {@link org.gbif.common.messaging.api.messages.DwcaValidationFinishedMessage } is received.
+ * ToDo Implement DwCAToAvro Conversion
+ */
+public class DwCAToAvroConverterCallBack extends AbstractMessageCallback<DwcaValidationFinishedMessage> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DwCAToAvroConverterCallBack.class);
+
+  @Override
+  public void handleMessage(DwcaValidationFinishedMessage dwcaValidationFinishedMessage) {
+    LOG.info("Received Download finished message, " + dwcaValidationFinishedMessage.toString());
```
Comment (fmendezh) [inline crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroConverterCallBack.java]: use LOG.info("Received Download finished message {}, dwcaValidationFinishedMessage); instead

File: crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroService.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  private final DwCAToAvroConfiguration configuration;
+  private MessageListener listener;
+
+  public DwCAToAvroService(DwCAToAvroConfiguration configuration) {
+    this.configuration = configuration;
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    LOG.info("Started dwca-to-avro service with parameters : " + configuration.toString());
```
Comment (fmendezh) [inline crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroService.java]: use something like LOG.info("Started dwca-to-avro service with parameters : {} ", configuration)); instead

File: crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroConverterCallBack.java (hunk 2 above)
Comment (aalbatross) [reply crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroConverterCallBack.java]: removed

File: crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroConverterCallBack.java (hunk 3 above)
Comment (aalbatross) [reply crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroConverterCallBack.java]: done

File: crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroService.java (hunk 4 above)
Comment (aalbatross) [reply crawler-cli/src/main/java/org/gbif/crawler/dwca/avroconverter/DwCAToAvroService.java]: done

**Pull request merged**